### ISSUE TITLE
Directories doesn't honour sourceSet include/exclude rules

### ISSUE THREAD

**Issue by gallardo** [kelemen/netbeans-gradle-project#236]
I'd be great if this plugin could also filter directories of source sets. For challenges implementing this functionality, see https://github.com/kelemen/netbeans-gradle-project/issues/184#issuecomment-136765260

**Comment by gallardo** [comment]
I began to implement a draft solution, but this is turning out to be too time consuming. I have opened a new issue, just for the record.

**Comment by kelemen** [comment]
I will add some rule of thumbs as soon as I can find the time to do it.

**Comment by JustGregory** [comment]
So, with the issues/discussions I've seen here and at @gradle/gradle about sourceSets, what are the full and final "rules of thumb" going to be in the end? How will additional rules affect any other gradle projects?

In other words, I have few issues with building projects with source paths, in my case the issues are in configuring/building a multiproject JavaFX. Additional rules of thumb won't break other things that work, will they?

**Comment by gallardo** [comment]
This issue report is about NB's project view, in concrete an aesthetics problem: some directories are listed even though they won't be considered by gradle for the compilation. As long as you run gradle tasks, the sanity of your build won't be affected at all.

**Comment by kelemen** [comment]
Yes, as Gallardo says, the goal is just not to list completely excluded directories. However, I will err on the side of caution, so I don't want a soltution which fails to list some non-excluded directories (i.e., I would rather list exculded empty directories).

**Comment by JustGregory** [comment]
Ah, okay. I was unnecessarily worried something would break in the next module release...
@gallardo summed it up correctly, and I see it's the project view that will get the tweaks. Thanks for that.
